core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -5529,9 +5529,23 @@ public static List mapValues(Map map) {
 
   /** Support the MAP_CONTAINS_KEY function. */
   public static Boolean mapContainsKey(Map map, Object key) {
+    for (Object mapKey : map.keySet()) {
+      if (isNumericEqual(mapKey, key)) {
+        return true;
+      }
+    }
     return map.containsKey(key);
   }
 
+  private static boolean isNumericEqual(Object obj1, Object obj2) {
+    if (obj1 instanceof Number && obj2 instanceof Number) {
+      double num1 = ((Number) obj1).doubleValue();
```

Comment:
this may not work for some decimal values which cannot be represented exactly as doubles
in general, it is not recommended to compare floating point values for equality, e.g, https://www.jetbrains.com/help/resharper/CompareOfFloatsByEqualityOperator.html

Reply:
you are right

Reply:
@mihaibudiu Should I add a layer of explicit conversion in case to convert the types in map and key to the biggest type

Reply:
Or convert BigDecimal for comparison, which one do you prefer?

Reply:
As I said, I don't think you should convert the values to double and then compare them.

Reply:
OK, I will convert the numeric type to double by default for comparison